Argo/Types/Decoded.swift

```diff
@@ -12,14 +12,23 @@ public enum Decoded<T> {
     default: return .None
     }
   }
+}
 
-  public static func optional<A>(p: Decoded<A>) -> Decoded<A?> {
-    switch p {
+public extension Decoded {
+  static func optional<T>(x: Decoded<T>) -> Decoded<T?> {
```

Comment:
This function isn't the inverse of `fromOptional` does that seem weird? Should we have a different name for this?